hail/src/main/scala/is/hail/expr/ir/IsConstant.scala

```diff
@@ -15,7 +15,7 @@ object CanEmit {
 object IsConstant {
   def apply(ir: IR): Boolean = {
     ir match {
-      case I32(_) | I64(_) | F32(_) | F64(_) | True() | False() | NA(_) | Literal(_, _) => true
+      case I32(_) | I64(_) | F32(_) | F64(_) | True() | False() | NA(_) | Str(_) | Literal(_, _) => true
```

Comment:
this change required to make the contig stuff work, so can't really stage this in a separate PR.